semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SemanticdbPipeline.scala

```diff
@@ -19,10 +19,14 @@ trait SemanticdbPipeline extends SemanticdbOps {
     def isIgnored: Boolean = {
       val matchesExtension = {
         val fileName = unit.source.file.name
-        fileName.endsWith(".scala") || fileName.endsWith(".sc") || fileName.endsWith(".java")
+        val lastDotIndex = fileName.lastIndexOf('.')
+        lastDotIndex > 0 && {
+          val ext = fileName.substring(lastDotIndex + 1)
+          SemanticdbPipeline.supportedExtensions.contains(ext)
+        }
       }
-      val matchesFilter = Option(unit.source.file).flatMap(f => Option(f.file))
-        .map(f => config.fileFilter.matches(f.getAbsolutePath)).getOrElse(true)
+      val matchesFilter = Option(unit.source.file)
```

Comment:
it's still a val and hence will be evaluated every time, including when `!matchesExtension`